### ISSUE TITLE
Backport 2953 to 2.x branch

### ISSUE THREAD

**Issue by metasim** [locationtech/geotrellis#3047]
See #2953.

**Comment by metasim** [comment]
I'm seeing this test fail on my branch off of ed5508e5b3e7899476d1623404a6023b7d2f6e89 but doesn't seem like it could be related (?).

```
[info] - should retrieve 5 bytes if requesting a large sequence 5 bytes from the end of the reader *** FAILED ***
[info]   java.nio.BufferUnderflowException:
[info]   at java.nio.HeapByteBuffer.get(HeapByteBuffer.java:151)
[info]   at java.nio.ByteBuffer.get(ByteBuffer.java:715)
[info]   at geotrellis.util.StreamingByteReader.getBytes(StreamingByteReader.scala:120)
[info]   at geotrellis.util.StreamingByteReaderSpec$$anonfun$1$$anonfun$apply$mcV$sp$9.apply(StreamingByteReaderSpec.scala:109)
[info]   at geotrellis.util.StreamingByteReaderSpec$$anonfun$1$$anonfun$apply$mcV$sp$9.apply(StreamingByteReaderSpec.scala:105)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FunSpecLike$$anon$1.apply(FunSpecLike.scala:454)
```

**Comment by pomadchin** [comment]
@metasim it is a bit hard to parse what you're talking about o:

**Comment by metasim** [comment]
@pomadchin  Sorry about the terseness. I had a few tests fail that didn't seem related to `RasterExtent`. I know the tests passed in `raster`, `vector`, and `spark`.

**Comment by pomadchin** [comment]
Added a CHANGELOG note, closing this issue. https://github.com/locationtech/geotrellis/commit/0460e4889d0d9d868006d8cf3ff9cff91e9c3450
